docs/adr/adr-014-bootstrap-from-previous-peers.md

```diff
@@ -0,0 +1,557 @@
+        return func(
+                lc fx.Lifecycle,
+                bpeers modp2p.Bootstrappers,
+                network modp2p.Network,
+                host host.Host,
+                conngater *conngater.BasicConnectionGater,
+                opts []p2p.Option[p2p.ClientParameters],
+        ) (libhead.Exchange[*header.ExtendedHeader], error) {
+                peers, err := cfg.trustedPeers(bpeers)
+                if err != nil {
+                        return nil, err
+                }
++              list, err := pstore.Load()
++               if err != nil {
++                       return nil, err
++               }
++               peers := append(peers, list...)
+                ids := make([]peer.ID, len(peers))
+                for index, peer := range peers {
+                        ids[index] = peer.ID
+                        host.Peerstore().AddAddrs(peer.ID, peer.Addrs, peerstore.PermanentAddrTTL)
+                }
+                exchange, err := p2p.NewExchange[*header.ExtendedHeader](host, ids, conngater, opts...)
+                if err != nil {
+                        return nil, err
+                }
+                lc.Append(fx.Hook{
+                        OnStart: func(ctx context.Context) error {
+                                return exchange.Start(ctx)
+                        },
+                        OnStop: func(ctx context.Context) error {
+                                return exchange.Stop(ctx)
+                        },
+                })
+                return exchange, nil
+        }
+ }
+```
+
+(_Code Snippet 4.a: Example of required changes to: celestia-node/nodebuilder/header/constructors.go_)
+
+</br>
+
+## Approach B: Periodic sampling of peers from the peerstore for "good peers" selection using liveness buckets
+
+In this approach, we will periodically sample peers from the host's `Peerstore` and select the "good peers", with the "good peers" being the ones that have been connected to the longest possible period of time.
+
+Taking inspiration from [this proposition](https://github.com/libp2p/go-libp2p-kad-dht/issues/254#issuecomment-633806123) we suggest to look for peers that have been continuously online, such that, for the given buckets of connection durations:
+
+* < 1 hour ago
+* < 1 day ago
+* < 1 week ago
+* < 1 month ago.
+
+the "good enough peers" to select should be present in most recent buckets, and should be peers we've seen often. For example, if a peer has been connected to the node for 1 day, then it should figure up in the buckets: "< 1 hour" and , "< 1 day", hence making it a "good enough peer" to select for persistence.
+
+The bucketing logic that triages peers into the mentioned buckets is executed periodically, with a new list of "good enough peers" being selected and persisted at that same time, and also at the moment of node shutdown.
+
+## Implementation
+
+### Storage
+
+We suggest to use the same storage mechanism proposed in approach A.
```

Comment:
Please note how often this key will be written to in this approach, because it is relevant for our storage backend